socketio/src/asynchronous/client/builder.rs

```diff
@@ -343,26 +385,70 @@ impl ClientBuilder {
     /// }
     /// ```
     pub async fn connect(self) -> Result<Client> {
+        let reconnect_delay_min = self.reconnect_delay_min;
+        let reconnect_delay_max = self.reconnect_delay_max;
+        let max_reconnect_attempts = self.max_reconnect_attempts;
+
         let socket = self.connect_manual().await?;
-        let socket_clone = socket.clone();
+        let mut socket_clone = socket.clone();
 
         // Use thread to consume items in iterator in order to call callbacks
         tokio::runtime::Handle::current().spawn(async move {
-            let mut stream = socket_clone.as_stream();
-            // Consume the stream until it returns None and the stream is closed.
-            while let Some(item) = stream.next().await {
-                if let e @ Err(Error::IncompleteResponseFromEngineIo(_)) = item {
-                    trace!("Network error occurred: {}", e.unwrap_err());
+            loop {
+                let mut stream = socket_clone.as_stream();
+                // Consume the stream until it returns None and the stream is closed.
+                while let Some(item) = stream.next().await {
+                    // Where does this get thrown by the async client? :confused:
```

Comment:
Ohh good catch! For the Async part, that would either be: 

- `IncompleteResponseFromReqwest`: Thrown for async polling transport defined [here](https://github.com/1c3t3a/rust-socketio/blob/57bcef7ad048d2e72e165ebb6d082dd2bf996ecf/engineio/src/asynchronous/async_transports/polling.rs)
- `WebsocketError`: Thrown for async Websocket transport defined [here](https://github.com/1c3t3a/rust-socketio/blob/57bcef7ad048d2e72e165ebb6d082dd2bf996ecf/engineio/src/asynchronous/async_transports/websocket_general.rs).
-